sim/battle.js

```diff
@@ -377,21 +377,23 @@ class Battle extends Dex.ModdedDex {
 		if (!source && this.event && this.event.target) source = this.event.target;
 		if (source === 'debug') source = this.p1.active[0];
 		status = this.getEffect(status);
-		if (this.pseudoWeather[status.id]) {
+
+		let pseudoWeather = this.pseudoWeather[status.id];
```

Comment:
This variable name should be `effectData` (or `pseudoWeatherData` if you really feel the need to be specific).

Reply:
Thanks, wasn't really sure.